functions/v2/label_gce_instance/main.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_label_gce_instance]
+import re
+
+from google.cloud import compute_v1
+from google.cloud.compute_v1.types import compute
+
+instances_client = compute_v1.InstancesClient()
+
+
+# CloudEvent function that labels newly-created GCE instances
+# with the entity (user or service account) that created them.
+#
+# @param {object} cloudevent A CloudEvent containing the Cloud Audit Log entry.
+# @param {object} cloudevent.data.protoPayload The Cloud Audit Log entry.
+def label_gce_instance(cloudevent):
+    # Extract parameters from the CloudEvent + Cloud Audit Log data
+    payload = cloudevent.data.get('protoPayload', dict())
+    auth_info = payload.get('authenticationInfo', dict())
+    creator = auth_info.get('principalEmail')
+
+    # Get relevant VM instance details from the cloudevent's `subject` property
+    # Example value:
+    #   compute.googleapis.com/projects/<PROJECT_ID>/zones/<ZONE_ID>/instances/<INSTANCE_NAME>
+    instance_params = cloudevent['subject'].split('/')
+
+    # Validate data
+    if not creator or not instance_params or len(instance_params) != 7:
+        # This is not something retries will fix, so don't throw an Exception
+        # (Thrown exceptions trigger retries *if* you enable retries in GCF.)
+        print('ERROR: Invalid event structure')
```

Comment:
nit: I would update this error message to be more specific like "principalEmail or Cloud Event subject not found in event"

Reply:
Done.